app/main.py

```diff
@@ -1,3 +1,21 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Any, Callable
+from functools import wraps
+
+
+def cache(func: Callable) -> Callable:
+    func_results = {}
+
+    @wraps(func)
+    def wrapper(*args) -> Any:
+        key = args
```

Comment:
It's not that essential to create an additional variable to store args, you can straight  use args for your check:
**if args in func_results:**
But the rest looks good.